urlbd77b745af_KokaKiwi_MCLauncher_tgz-pJ8-net_contrapunctus_lzma_BenchmarkJ8/src/com/kokakiwi/mclauncher/ui/simple/LoadingApplet.java

```diff
@@ -0,0 +1,247 @@
+                            + api.getUpdater().getCurrentFile().getSpeed()
+                            + " Kb/sec";
+                }
+            }
+            else if (api.getUpdater().getState() == State.EXTRACTING_PACKAGES
+                    && api.getUpdater().getCurrentFile() != null)
+            {
+                subtaskMessage = Translater.getString("updater.extracting")
+                        + ": "
+                        + api.getUpdater().getCurrentFile().getFileName() + " ";
+                subtaskMessage += api.getUpdater().getCurrentFile()
+                        .getPercentage()
+                        + "%";
+            }
+            
+            if (api.getUpdater().getError() != null)
+            {
+                msg = "Failed to launch";
+            }
+            
+            g.setFont(new Font(null, 1, 20));
+            FontMetrics fm = g.getFontMetrics();
+            g.drawString(msg, w / 2 - fm.stringWidth(msg) / 2,
+                    h / 2 - fm.getHeight() * 2);
+            
+            g.setFont(new Font(null, 0, 12));
+            fm = g.getFontMetrics();
+            msg = api.getUpdater().getState().getDescription();
+            if (api.getUpdater().getError() != null)
+            {
+                msg = api.getUpdater().getError();
+                subtaskMessage = "";
+            }
+            
+            g.drawString(msg, w / 2 - fm.stringWidth(msg) / 2,
+                    h / 2 + fm.getHeight() * 1);
+            msg = subtaskMessage;
+            g.drawString(msg, w / 2 - fm.stringWidth(msg) / 2,
+                    h / 2 + fm.getHeight() * 2);
+            
+            if (api.getUpdater().getError() == null)
+            {
+                g.setColor(Color.black);
+                g.fillRect(64, h - 64, w - 128 + 1, 5);
+                g.setColor(new Color(32768));
+                g.fillRect(64, h - 64, api.getUpdater().getPercentage()
+                        * (w - 128) / 100, 4);
+                g.setColor(new Color(2138144));
+                g.fillRect(65, h - 64 + 1, api.getUpdater().getPercentage()
+                        * (w - 128) / 100 - 2, 1);
+                
+                g.setColor(Color.LIGHT_GRAY);
+                g.setFont(new Font(null, 0, 12));
+                msg = api.getUpdater().getPercentage() + "%";
+                g.drawString(msg, w / 2 - fm.stringWidth(msg) / 2, h - 47);
+            }
+        }
+        
+        g.dispose();
+        g2.drawImage(img, 0, 0, w * 2, h * 2, null);
+    }
+    
+    
+    public void start()
+    {
+        final Thread t = new Thread(new Runnable() {
+            
+            public void run()
+            {
+                while (api.getUpdater().isRunning())
+                {
+                    repaint();
+                    try
+                    {
+                        Thread.sleep(10L);
+                    }
+                    catch (final InterruptedException e)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)